src/js/DatePicker.js

```diff
@@ -0,0 +1,436 @@
+const $ = window.$ || require("jquery");
+require('jquery-ui/ui/widgets/datepicker');
+
+class DatePicker {
+    /**
+     * DatePicker
+     * @param {DatePickerUtils} utils
+     * @param {DatePickerHighlightUtils} highlightUtils
+     * @param {DatePickerAccessibleLabelUtils} accessibleLabelUtils
+     * @param {DatePickerNavigationManager} navigationManager
+     */
+    constructor (utils, highlightUtils, accessibleLabelUtils, navigationManager) {
+        /**
+         * Root html element
+         * @type {JQuery|null}
+         */
+        this.$html = null;
+        /**
+         * Calendar trigger button
+         * @type {JQuery|null}
+         */
+        this.$triggerButton = null;
+        /**
+         * Date input
+         * @type {JQuery|null}
+         */
+        this.$dateInput = null;
+        /**
+         * Calendar next month link
+         * @type {JQuery|null}
+         */
+        this.$nextLink = null;
+        /**
+         * Calendar previous month link
+         * @type {JQuery|null}
+         */
+        this.$prevLink = null;
+        /**
+         * Calendar today link
+         * @type {JQuery|null}
+         */
+        this.$todayLink = null;
+        /**
+         * Calendar wrapper
+         * @type {JQuery|null}
+         */
+        this.$datePickerContainer = null;
+
+        this.utils = utils;
+        this.highlightUtils = highlightUtils;
+        this.accessibleLabelUtils = accessibleLabelUtils;
+        this.navigationManager = navigationManager;
+    }
+
+    /**
+     * Initialize
+     * @param {jQuery} $html
+     */
+    init ($html) {
+        this.$html = $html;
+
+        const $datePickers = $html.find('[data-pulsardatepicker="true"]');
+        let defaultDateFormat = 'dd/mm/yy';
+        let inputPlaceholder = 'dd/mm/yyyy';
```

Comment:
What's the difference between the placeholder and the date format? Shouldn't they be the same? Otherwise, it's a placeholder showing `dd/mm/yyyy` but with entered data, it shows `10/12/20`?

Reply:
The jQuery UI datepicker dateFormat uses `yy` for 4 digit format. `defaultDateFormat` is used for the `dateFormat` option. `inputPlaceholder` is used for the placeholder.